app/views/shared/_object_changes.html.haml

```diff
@@ -12,17 +12,37 @@
         - if can? :revert_attribute, version
           %th Action
     %tbody
-      - if version.event!= 'destroy'
+      - if version.event != 'destroy'
         - version.changeset.reject{ |_, values| values[0].blank? && values[1].blank? }.each do |attribute, values|
           %tr
-            %td= attribute
-            - if version.event != 'create'
-              %td= values[0].blank? ?  '-' : values[0]
-            %td= values[1].blank? ?  '-' : values[1]
+            - if attribute.include?('_id')
+              - model_name = attribute.chomp('_id').camelize
+              %td= attribute.chomp('_id')
+              - if version.event != 'create'
+                %td
+                  = object_last_description(model_name, values[0])
+                  = "(ID: #{values[0].blank? ?  '-' : values[0]})"
+              %td
+                = object_last_description(model_name, values[1])
```

Comment:
Same here.I've changed this to check if the associated object (track in this case) exists first and checks in versions only otherwise.